File: android/src/main/java/expo/modules/khalticheckout/constants/KhaltiConstants.kt
package expo.modules.khalticheckout.constants

/**
 * Constants used throughout the Khalti Payment SDK module.
 */
object KhaltiConstants {
  const val TAG = "KhaltiPaymentSDK"
  
  // Error codes
  const val ERROR_NO_ACTIVITY = "ERR_NO_ACTIVITY"
  const val ERROR_PAYMENT_INIT = "ERR_PAYMENT_INIT"
  const val ERROR_PAYMENT_FAILED = "ERR_PAYMENT_FAILED"
  const val ERROR_PAYMENT_CANCELED = "ERR_PAYMENT_CANCELED"
  const val ERROR_INVALID_CONFIG = "ERR_INVALID_CONFIG"
  const val ERROR_CLOSE_FAILED = "ERR_CLOSE_PAYMENT"
  const val ERROR_GET_CONFIG = "ERR_GET_CONFIG"

  // Environment constants
  const val ENV_PROD = "PROD"
  const val ENV_TEST = "TEST"
  
  // Event names
  const val EVENT_PAYMENT_SUCCESS = "onPaymentSuccess"
  const val EVENT_PAYMENT_ERROR = "onPaymentError"
  const val EVENT_PAYMENT_CANCEL = "onPaymentCancel"
}

File: ios/KhaltiConstants.swift
import Foundation

/**
 * Constants used throughout the Khalti Payment SDK module.
 */
struct KhaltiConstants {
    static let tag = "KhaltiPaymentSDK"
    
    // Error codes
    static let errorNoViewController = "ERR_NO_VIEW_CONTROLLER"
    static let errorPaymentInit = "ERR_PAYMENT_INIT"
    static let errorPaymentFailed = "ERR_PAYMENT_FAILED"
    static let errorPaymentCanceled = "ERR_PAYMENT_CANCELED"
    static let errorInvalidConfig = "ERR_INVALID_CONFIG"
    static let errorCloseFailed = "ERR_CLOSE_PAYMENT"
    static let errorGetConfig = "ERR_GET_CONFIG"
    
    // Environment constants
    static let envProd = "PROD"
    static let envTest = "TEST"
    
    // Event names
    static let eventPaymentSuccess = "onPaymentSuccess"
    static let eventPaymentError = "onPaymentError"
    static let eventPaymentCancel = "onPaymentCancel"
}
